File: src/main/java/se/feraswilson/automationservice/domain/ActionResult.java
package se.feraswilson.automationservice.domain;

public class ActionResult {
    private StatusCode statusCode;
    private String errorMsg;
    private String output;

    public ActionResult() {
        this.statusCode = StatusCode.SUCCESS;
    }

    public ActionResult(StatusCode statusCode) {
        this.statusCode = statusCode;
    }

    public ActionResult(StatusCode statusCode, String output, String errorMsg) {
        this.statusCode = statusCode;
        this.output = output;
        this.errorMsg = errorMsg;
    }

    public StatusCode getStatusCode() {
        return statusCode;
    }

    public void setStatusCode(StatusCode statusCode) {
        this.statusCode = statusCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }

    public String getOutput() {
        return output;
    }

    public void setOutput(String output) {
        this.output = output;
    }
}

File: backend/src/main/java/com/dopplertask/doppler/domain/ActionResult.kt
package com.dopplertask.doppler.domain

data class ActionResult(var statusCode: StatusCode, var output: String, var errorMsg: String, var outputType: OutputType, var isBroadcastMessage: Boolean) {
    constructor(statusCode: StatusCode, output: String, errorMsg: String) : this(statusCode, output, errorMsg, OutputType.STRING, true)
    constructor() : this(StatusCode.SUCCESS, "", "")
}